sdk/tests/deqp/modules/shared/glsBuiltinPrecisionTests.js

```diff
@@ -4609,6 +4616,17 @@ var setParentClass = function(child, parent) {
         return v3;
     };
 
+    glsBuiltinPrecisionTests.Mix.prototype.doExpandAlternative = function(ctx, args) {
+        // x + (y - x) * a
+        var x = args.a;
+        var y = args.b;
+        var a = args.c;
+        var v0 = app(new glsBuiltinPrecisionTests.Sub(), y, x);
+        var v1 = app(new glsBuiltinPrecisionTests.Mul(), a, v0);
+        var v2 = app(new glsBuiltinPrecisionTests.Add(), x, v1);
+        return v2;
+    };
+
```

Comment:
Native deqp has an alternative implementation for mix() builtin precision. They are:
1). ret0 = x * (1 - a) + y * a
2). ret1 = x + (y - x) * a. 
If the result of mix(x, y, a) from shader is in this range [min(ret0, ret1), max(ret0, ret1)], It will think that the result is correct. 

See the native deqp code https://android.googlesource.com/platform/external/deqp/+/deqp-dev/modules/glshared/glsBuiltinPrecisionTests.cpp#3268 and the class of Alternative::doApply and the definition of "|“ (union operation) in tcuInterval in native deqp: https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/common/tcuInterval.hpp#88 

However, in JS implementation, there is no "alternative" class at all. Considering that there are only a few users of "Alternative" class, and only mix() failed. For simplicity, I use another function to calculate mix() with method 2) above, then union the result with the original one (just like what it does in native deqp) in DerivedFunc.doApply().